src/app.rs

```diff
@@ -1144,6 +1155,16 @@ impl SearchState {
             self.last_search_field = self.search_field;
         }
 
+        // Invalidate when the whole word setting changes.
+        if self.whole_word != self.last_whole_word {
+            invalidate = true;
+            self.last_whole_word = self.whole_word;
+            if self.whole_word {
+                let regex_string = format!("\\b{}\\b", escape(&self.query));
+                self.last_word_regex = Some(Regex::new(regex_string.as_str()).unwrap());
```

Comment:
Oh, forgot to mention: I'm pretty sure `regex_string.as_str()` can be written `&regex_string`, because there's an impl somewhere for `&String` to behave like `&str` (again with those implied impls).